### PR TITLE
Withdrawal support

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: src/entities/mellow/config/utils.ts [hunk 1]
```diff
@@ -14,7 +13,7 @@ export const disableMaturedWeights = (config: NetworkConfiguration): NetworkConf
             return {
               ...vault,
               weight:
-                Date.now().valueOf() > vault.maturityTimestampMS - 48 * ONE_HOUR_IN_MS
+                Date.now().valueOf() + 48 * ONE_HOUR_IN_MS > vault.maturityTimestampMS
```
Comment (ssbarbee) [inline src/entities/mellow/config/utils.ts:16]: perhaps this 48h should be a constant of its own. I would prefer to change stuff in constants.ts if tomorrow we say 48 should be 72. WDYT?

File: src/entities/mellow/mellowLpRouter.ts [hunk 2]
```diff
@@ -215,6 +219,22 @@ class MellowLpRouter {
     return getTokenInfo(this.readOnlyContracts.token.address).decimals;
   }
 
+  public get depositable(): boolean {
+    const latestMaturity = this.metadata.vaults.reduce(
+      (latest, vault) => Math.max(latest, vault.maturityTimestampMS),
+      0,
+    );
+    return !this.metadata.deprecated && Date.now().valueOf() + 48 * ONE_HOUR_IN_MS < latestMaturity;
```
Comment (ssbarbee) [inline src/entities/mellow/mellowLpRouter.ts:227]: again the magic 48 :) lets move the entire thing to constatnts.ts, the 48*ONE_HOUR_IN_MS.

File: tests/mellowLpRouter.test.ts [hunk 3]
```diff
@@ -664,7 +714,44 @@ describe('Mellow Router Test Suite', () => {
...
+            {
+              weight: 25,
+              pools: ['Compound - ETH'],
+              maturityTimestampMS: 1670427875000,
+              estimatedHistoricApy: 0.0,
+            },
+            {
+              weight: 25,
+              pools: ['Compound - ETH'],
+              maturityTimestampMS: 1670427875000,
```
Comment (ssbarbee) [inline tests/mellowLpRouter.test.ts:737]: minor suggestion here. Can we perhaps use some date constructor and use the .valueOf here? Perhaps something like new Date('2022-01-01 23:59:59').valueOf() is more human readable. WDYT?

Comment (0xZenus) [reply tests/mellowLpRouter.test.ts:737]: This exact timestamp is associated with the underlying smart contract, so it's not hard-coded or so.

File: src/entities/mellow/config/config.ts [hunk 4]
```diff
@@ -39,6 +40,7 @@ const networkConfigurations: { [key: string]: NetworkConfiguration } = {
           show: true,
           soon: false,
           deprecated: false,
+          withdrawable: false,
```
Comment (avniculae) [inline src/entities/mellow/config/config.ts:43]: this should be per vault, not per router

File: src/entities/mellow/mellowLpRouter.ts [hunk 5]
```diff
@@ -215,6 +219,22 @@ class MellowLpRouter {
...
 
+  public get depositable(): boolean {
+    const latestMaturity = this.metadata.vaults.reduce(
+      (latest, vault) => Math.max(latest, vault.maturityTimestampMS),
+      0,
+    );
+    return !this.metadata.deprecated && Date.now().valueOf() + DEPOSIT_WINDOW < latestMaturity;
+  }
+
+  public get withdrawable(): boolean {
```
Comment (avniculae) [inline src/entities/mellow/mellowLpRouter.ts:230]: this should get the index of the vault to be withdrawn from, right?

Comment (0xZenus) [reply src/entities/mellow/mellowLpRouter.ts]: resolved.

File: src/entities/mellow/config/config.ts (hunk 4 above)
Comment (0xZenus) [reply src/entities/mellow/config/config.ts]: yes, resolved.

File: src/entities/mellow/mellowLpRouter.ts [hunk 6]
```diff
@@ -215,6 +219,25 @@ class MellowLpRouter {
     return getTokenInfo(this.readOnlyContracts.token.address).decimals;
   }
 
+  public get depositable(): boolean {
+    const latestMaturity = this.metadata.vaults.reduce(
+      (latest, vault) => Math.max(latest, vault.maturityTimestampMS),
+      0,
+    );
+    return !this.metadata.deprecated && Date.now().valueOf() + DEPOSIT_WINDOW < latestMaturity;
```
[the remark this answers is not in the document]
Comment (0xZenus) [reply src/entities/mellow/mellowLpRouter.ts]: https://github.com/Voltz-Protocol/v1-sdk/pull/129/commits/a9ed6adfb6e55a65623ab89cf70416b2f21f3479

File: tests/mellowLpRouter.test.ts (hunk 3 above)
Comment (0xZenus) [reply tests/mellowLpRouter.test.ts:737]: hmm, this need to coincide with the timestamp of the underlying maturity timestamp that can be found on Etherscan.

**Review by ssbarbee — APPROVED** [review]

**Pull request merged**